.github/workflows/ci.yaml

```diff
@@ -47,6 +47,30 @@ jobs:
 
       - name: Run tests
         run: yarn test
+  coverage:
+    runs-on: ubuntu-latest
+    env:
+      FOUNDRY_PROFILE: ci
+      ETH_RPC_URL: ${{ secrets.ETH_RPC_URL }}
+      INFURA_ID: ${{ secrets.INFURA_ID }}
+      ETHERSCAN_API_KEY: ${{ secrets.ETHERSCAN_API_KEY }}
+      POLYGONSCAN_API_KEY: ${{ secrets.POLYGONSCAN_API_KEY }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - uses: volta-cli/action@v1
+
+      - name: Install Foundry
+        uses: onbjerg/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - run: yarn install
+
+      - name: Run coverage
+        run: yarn coverage
```

Comment:
Build is failing because of an invalid mnemonic phrase is there something I need to do to setup the CI correctly? Things seem to be working locally

Reply:
From a high level on scope here:
- Coverage for the `umbra-js` and `contracts-periphery` packages would be great
- We can skip `contracts-core` since the core contracts aren't changing anymore and that package isn't really touched these days
- Frontend coverage is roughly zero so lets skip that too

Then when testing the current state  of things it seems:
- Running `yarn coverage` in the `umbra-js` directory worked
- Running `forge coverage` in the `contracts-periphery` directory worked, so we probably need to add an alias into the package.json there

The mnemonic issue you mentioned above seems to be for the `contracts-core` directory but we can skip that for coverage.

`forge coverage --report lcov` creates a coverage report at `./lcov.info`. I'm sure `nyc` can gnerate LCOV too since that's a standard format. So then maybe what we do is generate the two LCOV reports, filter out certain directories (such as `test/` in `contracts-periphery`), merge the two files then check for a minimum overall coverage threshold in CI? Here is how we filter out directories and check for minimum coverage threshold for a pure forge repo: https://github.com/ScopeLift/foundry-template/blob/f2b8c5f72a4c85533957537e804e6816d9744bc8/.github/workflows/ci.yml#L42-L68

Reply:
Ah wait but build is failing because of the missing mnemonic, so I just added one to github CI. I used `test test test test test test test test test test silent junk` based on this https://github.com/ScopeLift/umbra-protocol/blob/727d678ee30a42f656705b13bc7bb7904b3c36a3/contracts-core/scripts/deploy-registry.js#L58-L62

Reply:
I think this should be working with lcov now.

Reply:
Awesome, yea all the CI checks now run so marking this resolved!